packages/core/src/Message.ts

```diff
@@ -1,42 +1,31 @@
-import { Map } from 'immutable'
 import A from 'automerge'
-import { RepoSnapshot, RepoHistory } from './types'
+import { RepoSnapshot, RepoHistory, Clock, ClockMap } from './types'
 
-type Clock = Map<string, number>
-
-/**
- * Kick off our interaction with a peer by telling them how many documents we have
- */
+/** Kick off our interaction with a peer by telling them how many documents we have */
 export const HELLO = 'HELLO'
 export interface HelloMessage {
   type: typeof HELLO
   documentCount: number
 }
 
-/**
- * Request a document we don't have (snapshot and changes)
- */
+/** Request a document we don't have (snapshot and changes) */
 export const REQUEST_DOCS = 'REQUEST_DOCS'
 export interface RequestDocsMessage {
   type: typeof REQUEST_DOCS
   documentIds: string[]
 }
 
-/**
- * Advertise new document
- */
+/** Advertise new document */
 export const ADVERTISE_DOCS = 'ADVERTISE_DOCS'
 export interface AdvertiseDocsMessage {
   type: typeof ADVERTISE_DOCS
-  documents: {
+  clocks: {
```

Comment:
Change the name of this property to `clocks` since that's what we're sending, not documents.